src/NServiceBus.RavenDB/Timeouts/RavenDbTimeoutStorage.cs

```diff
@@ -43,6 +45,38 @@ protected override void Setup(FeatureConfigurationContext context)
 
             context.Container.ConfigureComponent(() => new TimeoutPersister(store), DependencyLifecycle.InstancePerCall);
             context.Container.ConfigureComponent(() => new QueryTimeouts(store, context.Settings.EndpointName().ToString()), DependencyLifecycle.SingleInstance); // Needs to be SingleInstance because it contains cleanup state
+            context.RegisterStartupTask(builder =>
```

Comment:
Why does this need a builder access? The query is a single instance which could be created in the Setup of the feature and you could then pass it in over the ctor into the startup task directly